include/envoy/request_info/request_info.h

```diff
@@ -68,66 +68,134 @@ class RequestInfo {
    */
   virtual void onUpstreamHostSelected(Upstream::HostDescriptionConstSharedPtr host) PURE;
 
+  /**
+   * @return the number of body bytes received in the request.
+   */
+  virtual uint64_t bytesReceived() const PURE;
+
+  /**
+   * @return the protocol of the request.
+   */
+  virtual const Optional<Http::Protocol>& protocol() const PURE;
```

Comment:
As long as you are moving a lot of stuff around in here, can you switch this to not return by reference? It will be faster. Same for `responseCode()` and anything else that is an optional of a small primitive type.